tests/base.py

```diff
@@ -0,0 +1,158 @@
+        elif error_type == "z":
+            parity = 2 * (indx % 2) - 1  # Even is -1, Odd is 1
+            neighbors = [
+                (0.0, row - parity * 0.5, col - 0.5),
+                (0.0, row + parity * 0.5, col + 0.5),
+            ]
+            return [x for x in neighbors if x[2] > 0 and x[2] < dw - 1]
+        return []
+
+    def simulate(self, error, i, logical_readout_type):
+        # Set up circuit
+        qubit = self.encoder_type(self.params)
+        if logical_readout_type == "Z":
+            qubit.reset_z()
+        else:
+            qubit.reset_x()
+        qubit.stabilize()
+        qubit.circ.__getattribute__(error)(qubit.lattice.qregisters["data"][i])  # error
+        qubit.stabilize()
+        if logical_readout_type == "Z":
+            qubit.readout_z()
+        else:
+            qubit.readout_x()
+
+        # Simulate
+        results = (
+            execute(qubit.circ, Aer.get_backend("aer_simulator"), shots=1000,)
+            .result()
+            .get_counts()
+        )
+
+        # Get Logical Error Rate
+        logical_error_rate = self.get_logical_error_rate(
+            results, 0, logical_readout_type
+        )
+        print(
+            f"Decoding result for {error} Error on the {i}th data qubit ({logical_readout_type} readout), logical_error_rate: {logical_error_rate}."
+        )
+        self.assertEqual(
+            logical_error_rate,
+            0,
+            f"Decoding did not work for an {error} Error on the {i}th data qubit.",
+        )
+
+        # Check if syndromes were parsed correctly
+        expected_neighbors = self.get_neighbors(i, error)
+        print(f"Expected neighbors: {expected_neighbors}")
+
+        readout_strings = list(results.keys())
+        for readout_string in readout_strings:
+            _, processed_results = self.decoder.parse_readout(readout_string)
+            nodes = sum(list(processed_results.values()), [])
+            for x in nodes:
+                self.assertIn(x, expected_neighbors)
+
+    def test_single_errors(self):
+        """
+        Setting up a |+z> state, stabilizing twice, and reading out logical Z.
+        Inserting X and Z gates on each data qubit between tbe two stabilizer measurement rounds.
+
+        Then, testing:
+        1.  The MWPM decoder is able to correct these single qubit errors
+            and predict the correct logical readout value.
+        2.  Checking if the _string2node readout string to
+            syndrome node parser is working correctly.
+        """
+        print(
+            "\n"
+            + 9 * "="
+            + "\n"
+            + self.__class__.__name__
+            + ": "
+            + str(self.params)
+            + "\n"
+        )
+        # set up circuit
+        d = self.params["d"]
+        for i in range(d[constants.DH] * d[constants.DW]):
+            for error in self.errors:
+                if error == "x" and d[constants.DH] == 1:
```

Comment:
Good catch!

As you already saw, for a single row surface code (e.g. phase-protected rep code), an x flip on any qubit (i.e. a full column) is equivalent to a logical X flip. Same for z flips on a single column surface code.